File: Easy/Common elements/common-elements.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution
{
    public:    
       vector <int> commonElements (int A[], int B[], int C[], int n1, int n2, int n3)
     {
            //code here.
            map<int,int>mp1,mp2,mp3;
            for(int i=0;i<n1;i++)mp1[A[i]]++;
            for(int i=0;i<n2;i++){
                if(mp1[B[i]])mp2[B[i]]++;
            }
            for(int i=0;i<n3;i++){
                if(mp2[C[i]])mp3[C[i]]++;
            }
            vector<int>ans;
            for(auto pair:mp3)ans.push_back(pair.first);
            return ans;
        }

};

//{ Driver Code Starts.

int main ()
{
    int t; cin >> t;
    while (t--)
    {
        int n1, n2, n3; 
        cin >> n1 >> n2 >> n3;
        int A[n1];
        int B[n2];
        int C[n3];
        
        for (int i = 0; i < n1; i++) cin >> A[i];
        for (int i = 0; i < n2; i++) cin >> B[i];
        for (int i = 0; i < n3; i++) cin >> C[i];
        
        Solution ob;
        
        vector <int> res = ob.commonElements (A, B, C, n1, n2, n3);
        if (res.size () == 0) 
            cout << -1;
        for (int i = 0; i < res.size (); i++) 
            cout << res[i] << " "; 
        cout << endl;
    }
}
// } Driver Code Ends

File: Easy/Common elements/common-elements.py
#User function Template for python3

class Solution:
    def commonElements (self,A, B, C, n1, n2, n3):
                                        s1 = set(A)
                                        s2 = set(B)
                                        s3 = set(C)
                                         
                                       
                                        set1 = s1.intersection(s2)         
                                        result_set = set1.intersection(s3)
                                         
                                        final_list = list(result_set)
                                        final_list.sort()
                                        return final_list


#{ 
 # Driver Code Starts
#Initial Template for Python 3


t = int (input ())
for tc in range (t):
    n1, n2, n3 = list(map(int,input().split()))
    A = list(map(int,input().split()))
    B = list(map(int,input().split()))
    C = list(map(int,input().split()))
    
    ob = Solution()
    
    res = ob.commonElements (A, B, C, n1, n2, n3)
    
    if len (res) == 0:
        print (-1)
    else:
        for i in range (len (res)):
            print (res[i], end=" ")
        print ()

# } Driver Code Ends